src/com/goide/runconfig/testing/coverage/GoCoverageProgramRunner.java

```diff
@@ -68,13 +124,46 @@ protected RunContentDescriptor doExecute(@NotNull RunProfileState state, @NotNul
     }
     FileDocumentManager.getInstance().saveAllDocuments();
     CoverageEnabledConfiguration coverageEnabledConfiguration = CoverageEnabledConfiguration.getOrCreate(runConfiguration);
-    runningState.setCoverageFilePath(coverageEnabledConfiguration.getCoverageFilePath());
+    CoveragePatcher runningStatePatcher = new CoveragePatcher(coverageEnabledConfiguration, runConfiguration);
+    runningState.setPatcher(runningStatePatcher);
+
+    if (runConfiguration.getKind() == GoTestRunConfiguration.Kind.DIRECTORY &&
+        !runningStatePatcher.hasPackageCoverageExecutable()) {
+      promptForPackageCoverageTool(runConfiguration);
+    }
 
     ExecutionResult executionResult = state.execute(environment.getExecutor(), this);
     if (executionResult == null) {
       return null;
     }
+    executionResult.getProcessHandler().addProcessListener(new GoCoverageProcessListener(
+        runConfiguration.getDirectoryPath(),
```

Comment:
So all reports are generated right in user's sources?
Sorry, this is not acceptable.

1. IDE must not pollute user's sources files with other unrelated files
2. IDE should not overwrite already existing files by invoking Coverage
3. Recursing iterating across all directory files while coverage merging will be too long because of iterating the files that have nothing to do with coverage